components/Connect.tsx

```diff
@@ -70,6 +70,46 @@ export default function BluetoothConnect({ bikeCredentials, setBikeInstance, bac
                     Back to login
                 </Button>
             </P>
+
+            <P vertical={10}>
+                <Button
+                    onClick={() => {
+                        const blob = new Blob([JSON.stringify(bikeCredentials)], { type: 'application/json' })
+                        const url = URL.createObjectURL(blob)
+
+                        // If the user is on iOS and has the Bluefy browser, we can't save the file directly
+                        // so we'll just show the user the JSON file and they can save it manually
+                        if (navigator.userAgent.includes('iPhone') && navigator.userAgent.includes('Bluefy')) {
+                            // Show the user the JSON on the same page in a scrollable div
+                            const div = document.createElement('div')
+
+                            const pre = document.createElement('pre')
+
+                            const code = document.createElement('code')
+
+                            code.innerText = JSON.stringify(bikeCredentials, null, 2)
+
+                            pre.appendChild(code)
+
+                            div.appendChild(pre)
```

Comment:
We are using react there is no need to create html elements by hand.
You can easily add a state to this component that contains the credentials and only show the credentials if it contains content.